smithy-codegen-core/src/test/java/software/amazon/smithy/codegen/core/writer/CodegenWriterTest.java

```diff
@@ -113,4 +113,18 @@ public void importsUseReferencesFromSymbols() {
         assertThat(writer.getImportContainer().imports, hasKey("MyString"));
         assertThat(writer.getImportContainer().imports.get("MyString"), equalTo("java.lang.String"));
     }
+
+    @Test
+    public void canComposeSetWithSection() {
+        String testSection = "testSection";
+        MyWriter writer = new MyWriter("foo");
+
+        writer.onSection(testSection, text -> writer.write(text + "1, "));
+        writer.onSection(testSection, text -> writer.write(text + "2, "));
+        writer.onSection(testSection, text -> writer.write(text + "3"));
+
+        writer.write("[${L@testSection}]", "");
+
+        assertThat(writer.toString(), equalTo("[1, 2, 3]"));
```

Comment:
Oh, looks like this needs either a trailing newline, or you need to disable newlines in the writer

Reply:
I've pushed up a fix